app/users/search-users/searchUserInteractor.js

```diff
@@ -0,0 +1,40 @@
+const ValidationError = require('../../errors/validationError');
+const NotFound = require('../../errors/notFound');
+
+class SearchUserInteractor {
+    constructor ({userRepository, validator, presenter}) {
+        this.userRepository = userRepository;
+        this.validator = validator;
+        this.presenter = presenter;
+    }
+
+    async execute({id, name}) {
+        const errors = this.validator.validate({id, name});
+        if (errors.length > 0) {
+            this.presenter.presentFailure(new ValidationError(errors));
+            return;
+        }
+
+        if (errors.length > 0) {
+            this.presenter.presentFailure(new ValidationError(errors));
+            return;
+        }
+
+        let users;
+
+        if (Array.isArray(id)) {
+            users = await this.userRepository.findAll({ids: id, name});
+        } else {
+            users = await this.userRepository.findAll({ids: [id], name});
+        }
+
+        if (users.length === 0) {
+            this.presenter.presentFailure(new NotFound('No results.'));
+            return;
+        }
```

Comment:
1. `const userID = Array.isArray(id) ? id : [id];`
2. `users.length === 0` is definitely not an error) Its completely OK when your search didn't find anything